rust/template/d3log/src/json_framer.rs

```diff
@@ -0,0 +1,96 @@
+// this module pre-frames json objects out of an arbitrarily chunked byte string.
+// it is a really terrible idea to temporarily work around some limtations
+// with serde. specifically that it can read chunked blocks and a sequence
+// of self-framed json objects, but not both
+//
+// haven't been able to understand it yet but
+// https://github.com/TimelyDataflow/timely-dataflow/blob/master/timely/src/dataflow/operators/to_stream.rs#L73
+// may lead to a better answer.
+//
+// if it were going to stay Lexement should really just be the leading group
+// character
```

Comment:
I'm confused about this module in general, the linked code doesn't say anything about strings or utf8 and I'm not really even sure what the code actually does